ios/RNZoomUs.m

```diff
@@ -600,14 +600,161 @@ - (void)onClickShareScreen:(UIViewController *)parentVC {
+  MobileRTCMeetingService *ms = [[MobileRTC sharedRTC] getMeetingService];
+
+  if ([ms isMyself:userID]) {
+    MobileRTCMeetingUserInfo *userInfo = [ms userInfoByID:[ms myselfUserID]];
+
+    [self sendEventWithName:@"MeetingEvent" params:@{
+      @"event": @"myVideoStatusChanged",
+      @"userRole": [self getUserRole:[userInfo userRole]],
+      @"audioType": @([[userInfo audioStatus] audioType]),
+      @"isTalking": @([[userInfo audioStatus] isTalking]),
+      @"isMutedAudio": @((BOOL)([[userInfo audioStatus] audioType] == 2 ? YES : [[userInfo audioStatus] isMuted])),
+      @"isMutedVideo": @((BOOL)![[userInfo videoStatus] isSending]),
+    }];
+  }
+}
+
+// TODO: This one is not getting called
+- (void)onSinkMeetingVideoRequestUnmuteByHost:(void (^)(BOOL Accept))completion {
+  [self sendEventWithName:@"MeetingEvent" event:@"askUnMuteVideo"];
+
+  if (completion) {
+    completion(YES);
+  }
+}
+
+- (void)onSinkMeetingActiveVideo:(NSUInteger)userID {}
+
+- (void)onSinkMeetingVideoStatusChange:(NSUInteger)userID {}
+
+- (void)onMyVideoStateChange {}
+
+- (void)onSpotlightVideoChange:(BOOL)on {}
+
+- (void)onSinkMeetingPreviewStopped {}
+
+- (void)onSinkMeetingActiveVideoForDeck:(NSUInteger)userID {}
+
+- (void)onSinkMeetingVideoQualityChanged:(MobileRTCNetworkQuality)qality userID:(NSUInteger)userID {}
+
+- (void)onSinkMeetingShowMinimizeMeetingOrBackZoomUI:(MobileRTCMinimizeMeetingState)state {}
+
+
+#pragma mark - MobileRTCAudioServiceDelegate
+
+- (void)onSinkMeetingMyAudioTypeChange {
+  MobileRTCMeetingService *ms = [[MobileRTC sharedRTC] getMeetingService];
+  MobileRTCMeetingUserInfo *userInfo = [ms userInfoByID:[ms myselfUserID]];
+
+  [self sendEventWithName:@"MeetingEvent" params:@{
+    @"event": @"myAudioTypeChanged",
+    @"userRole": [self getUserRole:[userInfo userRole]],
+    @"audioType": @([[userInfo audioStatus] audioType]),
+    @"isTalking": @([[userInfo audioStatus] isTalking]),
+    @"isMutedAudio": @((BOOL)([[userInfo audioStatus] audioType] == 2 ? YES : [[userInfo audioStatus] isMuted])),
+    @"isMutedVideo": @((BOOL)![[userInfo videoStatus] isSending]),
+  }];
+}
+
+- (void)onSinkMeetingAudioStatusChange:(NSUInteger)userID audioStatus:(MobileRTC_AudioStatus)audioStatus {
+  MobileRTCMeetingService *ms = [[MobileRTC sharedRTC] getMeetingService];
+
+  if ([ms isMyself:userID]) {
+    MobileRTCMeetingUserInfo *userInfo = [ms userInfoByID:[ms myselfUserID]];
+
+    [self sendEventWithName:@"MeetingEvent" params:@{
+      @"event": @"myAudioStatusChanged",
+      @"userRole": [self getUserRole:[userInfo userRole]],
+      @"audioType": @([[userInfo audioStatus] audioType]),
+      @"isTalking": @([[userInfo audioStatus] isTalking]),
+      @"isMutedAudio": @((BOOL)([[userInfo audioStatus] audioType] == 2 ? YES : [[userInfo audioStatus] isMuted])),
+      @"isMutedVideo": @((BOOL)![[userInfo videoStatus] isSending]),
+    }];
```

Comment:
this repeats 3 times. on android we have
```
private void sendEvent(String name, String event, InMeetingUserInfo userInfo) {
```
So it would be great to have such abstraction on iOS, too

Reply:
Perfect, I was thinking about it but probably to tired to think about how to make it nicer.